mim/utils/mmpack/pack_cfg.py

```diff
@@ -0,0 +1,302 @@
+        else:
+            cfg = Config.fromfile(cfg)
+
+    origin_cfg = copy.deepcopy(cfg)
+    default_scope = cfg.get('default_scope', 'mmengine')
+    signal.signal(signal.SIGINT, lambda sig, frame: keyboardinterupt_handler(
+        sig, frame, export_root_dir))  # type: ignore[arg-type]
+
+    # automatically generate export_root_dir
+    if export_root_dir is None:
+        export_root_dir = f'pack_from_{default_scope}_' + datetime.now(
+        ).strftime(r'%Y%m%d_%H%M%S')
+
+    # export_module_dir
+    export_module_dir = export_root_dir + '/pack'
+    if osp.exists(export_module_dir):
+        shutil.rmtree(export_module_dir)
+
+    # export config
+    if '.mim/' in cfg.filename:
+        cfg_dir = osp.join(export_module_dir,
+                           osp.dirname(cfg.filename.split('.mim/')[-1]))
+        cfg_pth = osp.join(export_module_dir, cfg.filename.split('.mim/')[-1])
+    else:
+        cfg_dir = osp.join(export_module_dir, 'configs')
+        cfg_pth = osp.join(cfg_dir, cfg.filename.split('/')[-1])
+
+    mkdir_or_exist(cfg_dir)
+
+    # NOTE: change parameters for fasterfast_ testing
+    if fast_test:
+        # for batch_norm using at least 2 data
+        if 'dataset' in cfg.train_dataloader.dataset:
+            cfg.train_dataloader.dataset.dataset.indices = [0, 1]
+        else:
+            cfg.train_dataloader.dataset.indices = [0, 1]
+        cfg.train_dataloader.batch_size = 2
+
+        if cfg.get('test_dataloader') is not None:
+            cfg.test_dataloader.dataset.indices = [0, 1]
+            cfg.test_dataloader.batch_size = 2
+
+        if cfg.get('val_dataloader') is not None:
+            cfg.val_dataloader.dataset.indices = [0, 1]
+            cfg.val_dataloader.batch_size = 2
+
+        if (cfg.train_cfg.get('type') == 'IterBasedTrainLoop') \
+                or (cfg.train_cfg.get('by_epoch') is None
+                    and cfg.train_cfg.get('type') != 'EpochBasedTrainLoop'):
+            cfg.train_cfg.max_iters = 2
+        else:
+            cfg.train_cfg.max_epochs = 2
+
+        cfg.train_cfg.val_interval = 1
+        cfg.default_hooks.logger.interval = 1
+
+        if 'param_scheduler' in cfg and cfg.param_scheduler is not None:
+            if isinstance(cfg.param_scheduler, list):
+                for lr_sc in cfg.param_scheduler:
+                    lr_sc.begin = 0
+                    lr_sc.end = 2
+            else:
+                cfg.param_scheduler.begin = 0
+                cfg.param_scheduler.end = 2
+
+    cfg.dump(cfg_pth)
+
+    _replace_config_scope_to_pack(cfg_pth)
+
+    # transform to default_scope
+    init_default_scope(default_scope)
+
+    # wrap ``Registry.build()`` for exporting modules
+    _wrapper_all_registries_build_func(
+        export_module_dir=export_module_dir, scope=default_scope)
+
+    cfg['work_dir'] = osp.join(export_root_dir,
+                               'export_log')  # creat temp work_dirs for export
+
+    def error_postprocess():
```

Comment:
Since we've defined the signal handler outside of the function, it could be better to move this function outside of it too.